lib/Runtime/Library/JavascriptFunction.cpp

```diff
@@ -1126,8 +1126,24 @@ namespace Js
     template Var JavascriptFunction::CallFunction<false>(RecyclableObject* function, JavascriptMethod entryPoint, Arguments args);
 
 #ifdef _M_IX86
-    template <bool doStackProbe>
-    Var JavascriptFunction::CallFunction(RecyclableObject* function, JavascriptMethod entryPoint, Arguments args)
+#ifdef __clang__
+void __cdecl _alloca_probe_16()
+{
+    // todo: fix this!!!
+    __asm
+    {
+        push    ecx
+        lea     ecx, [esp + 8]
+        sub     ecx, eax
+        and     ecx, (16 - 1)
+        add     eax, ecx
+        ret
```

Comment:
Confused by what this is doing but I guess that's why there is a todo. I'm assuming the fix involves changing esp. Right now, it looks like this would leave the stack unaligned since you do a push ecx but no pop?